### PR TITLE
Kp/sliding window

### PR DESCRIPTION
**Pull request by kyra-at-ada**
Summary of Changes:
- Add overview of sliding window technique
- How to determine if a problem can use the sliding window technique
- Applying a static sliding window to a problem
- Applying a dynamic sliding window to a problem

For review:
- typos (since they're finding all of mine lol)
- clarity of content
- Is it too long? I'm wondering if we should make some cuts to make the overall topics shorter

Notes:
- still need to make the video

### CODE REVIEW

File: 07-algorithms/05-sliding-window.md [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+Try using the algorithm we developed for the problem above on this input array: `[-1, -4, 0, 5, 3, 2, 1]`. Assume the `target` is 5. The sliding window assumes that the sum of the window increases with each shift to the right, but because of the presence of negative numbers in some cases the sum decreases. The right edge of our window will end up past the element at index 3, `5` before we ever start shifting the left edge of our window. Thus we will never catch index 3 as its own valid subarray adding up to the target value.
+
+<break>
+
+It is possible to modify the sliding window technique to work with non-positives, but Kadane's algorithm is easier and more efficient!
+### !end-callout
+
+## Summary
+
+The sliding window technique is a useful technique that helps us solve optimization problems that ask us to find a contiguous subsequence within a larger sequence. Note that the dynamic sliding window technique often doesn't work in problems where sums and negative numbers are involved.
```
Comment (char-adadev) [inline 07-algorithms/05-sliding-window.md:429]:
```suggestion
The sliding window technique is a useful technique that helps us solve optimization problems that ask us to find a contiguous subsequence within a larger sequence. Note that the dynamic sliding window technique often doesn't work on problems where sums and negative numbers are involved.
```

File: 07-algorithms/05-sliding-window.md [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+
+Our sliding window can be static means that it stays a fixed size as it progresses through the sequence or dynamic meaning that the right and left edges of our window can progress at different rates.
```
Comment (char-adadev) [inline 07-algorithms/05-sliding-window.md:431]:
```suggestion
Our sliding window can be static meaning that it stays a fixed size as it progresses through the sequence or dynamic meaning that the right and left edges of our window can progress at different rates.
```

**Review by char-adadev — APPROVED** [review]
Great work here! I left a few suggestions on wording and caught one potentially unfinished sentence. 

The only thing that really jumped out at me was the inclusion of a hard leetcode question as an example. I hope folx don't assume we want them to go and solve all of the example questions! It might be worth adding a message that we don't intend for them to go and completely understand the algorithm for how each of the example problems work, but they are of course invited to follow their curiosity, as always.

**Comment by kyra-at-ada** [comment]
> Great work here! I left a few suggestions on wording and caught one potentially unfinished sentence.
> 
> The only thing that really jumped out at me was the inclusion of a hard leetcode question as an example. I hope folx don't assume we want them to go and solve all of the example questions! It might be worth adding a message that we don't intend for them to go and completely understand the algorithm for how each of the example problems work, but they are of course invited to follow their curiosity, as always.

I hear that, I was trying to incorporate feedback that folks wanted examples of what problems to work on including difficult questions so I included an easy, medium, and hard question. I can include a disclaimer

**Pull request merged**